components/brave_shields/browser/ad_block_service.cc

```diff
@@ -204,26 +204,15 @@ absl::optional<base::Value> AdBlockService::HiddenClassIdSelectors(
     custom_selectors = std::move(subscription_selectors);
   }
 
-#if !defined(OS_ANDROID) && !defined(CHROME_OS)
-  if (!base::FeatureList::IsEnabled(
-          brave_shields::features::kBraveAdblockCosmeticFilteringNative)) {
-    auto result_list = std::make_unique<base::ListValue>();
-    if (hide_selectors && hide_selectors->is_list()) {
-      result_list->Append(std::move(*hide_selectors));
-    }
-    if (custom_selectors && custom_selectors->is_list()) {
-      result_list->Append(std::move(*custom_selectors));
-    }
-
-    return std::move(*result_list);
-  }
-#endif
-
   if (!hide_selectors || !hide_selectors->is_list())
     hide_selectors = base::ListValue();
 
-  if (custom_selectors && custom_selectors->is_list())
-    hide_selectors->Append(std::move(*custom_selectors));
```

Comment:
why did we used to append a list?..

Reply:
great question :sweat_smile: 

I don't think it was correct. We had a couple of changes there when adding support for both backends in parallel, and I think something broke in the meantime. I think https://github.com/brave/brave-browser/issues/17013 was caused by this, but I've tried manually and it's fixed with these changes. I'll look into getting test coverage for it.